modules/auxiliary/scanner/http/xorcom_completepbx_diagnostics_file_read.rb

```diff
@@ -0,0 +1,190 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Xorcom CompletePBX Arbitrary File Read and Deletion via systemDataFileName',
+        'Description' => %q{
+          This module exploits an authenticated path traversal vulnerability in
+          Xorcom CompletePBX <= 5.2.35. The issue occurs due to improper validation of the
+          `systemDataFileName` parameter in the `diagnostics` module, allowing authenticated attackers
+          to retrieve arbitrary files from the system.
+
+          Additionally, the exploitation of this vulnerability results in the **deletion** of the
+          requested file from the target system.
+
+          The vulnerability is identified as CVE-2025-30005.
+        },
+        'Author' => ['Valentin Lobstein'],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2025-30005'],
+          ['URL', 'https://www.xorcom.com/products/completepbx/'],
+          ['URL', 'https://chocapikk.com/posts/2025/completepbx/']
+        ],
+        'DisclosureDate' => '2025-03-02',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE, OS_RESOURCE_LOSS],
+          'SideEffects' => [IOC_IN_LOGS],
+          'Reliability' => []
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETURI', [true, 'Base path of the CompletePBX instance', '/']),
+        OptString.new('USERNAME', [true, 'Username for authentication', 'admin']),
+        OptString.new('PASSWORD', [true, 'Password for authentication', 'admin']),
```

Comment:
The password is unlikely to be `admin` and so we should leave the default unset
```suggestion
        OptString.new('PASSWORD', [true, 'Password for authentication' ]),
```